benchmarks/helper.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+
+"""
+A wrapper over the benchmark infrastructure to generate commonly use commands,
+parse results and generate csv/graphs.
+
+The script works on manually written TABLE (see below). We can add more commands
+in the future.
+
+One example usage is
+-> python benchmarks/helper.py --suites=torchbench --inference
+This command will generate the commands for the default compilers (see DEFAULTS
+below) for inference, run them and visualize the logs.
+
+If you want to just run the commands, you could use the following command
+-> python benchmarks/helper.py --print_run_commands --suites=torchbench --inference
+
+Similarly, if you want to just visualize the already run logs
+-> python benchmarks/helper.py --visualize_logs --suites=torchbench --inference
+
+If you want to test float16
+-> python benchmarks/helper.py --suites=torchbench --inference --dtypes=float16
+
+"""
+
+import argparse
+import itertools
+import os
+
+import matplotlib.pyplot as plt
+import pandas as pd
+from matplotlib import rcParams
+from tabulate import tabulate
+
+rcParams.update({"figure.autolayout": True})
+plt.rc("axes", axisbelow=True)
+
+DEFAULT_OUTPUT_DIR = "benchmark_logs"
+
+
+TABLE = {
+    "training": {
+        "ts_nnc": "--training --speedup-ts --use-eval-mode --isolate",
```

Comment:
i'm curious- what does ts_nnc do? does it just compile forward graph and not compile backward?

or it uses torchscript autodiff and relies on that for bwd?

Reply:
Latter. Torchscript autodiff. Torchscript is used to capture the fwd and bwd graphs.
NNC or NVFuser are the backend compilers once the graphs are captured by torchscript.

Reply:
This does not use TorchDynamo to extract the graphs though. Hmm, I actually meant to use that.

I will add an option in common.py and correspondingly update this file in next PR.